src/plugins/echo_plugin/echo_plugin.cc

```diff
@@ -222,21 +204,23 @@ void EchoPlugin::RegisterEchoChannel() {
 
   for (const auto& topic_meta_itr : topic_meta_list) {
     const auto& topic_meta = topic_meta_itr.second;
-    EchoFunc echo_func;
-    if (options_.executor.empty()) {
-      echo_func = [this, echo_type{topic_meta.echo_type}](
-                      MsgWrapper& msg_wrapper) {
-        Echo(msg_wrapper, echo_type);
-      };
-    } else {
-      echo_func = [this, echo_type{topic_meta.echo_type}](
-                      MsgWrapper& msg_wrapper) {
-        executor_.Execute([this, msg_wrapper{std::move(msg_wrapper)}, echo_type]() mutable {
-          Echo(msg_wrapper, echo_type);
-        });
-      };
-    }
-
+    EchoFunc echo_func = [this, echo_type{topic_meta.echo_type}](MsgWrapper& msg_wrapper) {
+      auto buffer_view_ptr = aimrt::runtime::core::channel::TrySerializeMsgWithCache(msg_wrapper, echo_type);
+      if (!buffer_view_ptr) [[unlikely]] {
+        AIMRT_ERROR("Can not serialize msg type '{}' with echo type '{}'.",
+                    msg_wrapper.info.msg_type, echo_type);
+        return;
+      }
+      if (buffer_view_ptr->Size() == 1) {
+        const char* data = static_cast<const char*>(buffer_view_ptr->Data()[0].data);
+        AIMRT_INFO("\n{}\n---------------\n", std::string_view(data, buffer_view_ptr->Data()[0].len));
+      } else if (buffer_view_ptr->Size() > 1) {
+        AIMRT_INFO("\n{}\n---------------\n", buffer_view_ptr->JoinToString());
+      } else {
+        AIMRT_ERROR("Invalid buffer, topic_name: {}, msg_type: {}", msg_wrapper.info.topic_name, msg_wrapper.info.msg_type);
+      }
+    };
+    
     auto finditr = type_support_map_.find(topic_meta.msg_type);
     AIMRT_CHECK_ERROR_THROW(finditr != type_support_map_.end(),
                             "Can not find type '{}' in any type support pkg!", topic_meta.msg_type);
```

Comment:
Is the semantics of this location overlapping with the `get_type_support_func` above?  It's suggested that `get_type_support_func` be designed more generally, referenced in all locations needing type lookup, to maintain consistency.

Reply:
I believe this function design like this may be more clear , as it only performs a duplicate type check in type_support_map_.